packages/back-end/src/services/context.ts

```diff
@@ -90,50 +96,26 @@ export class ReqContextClass implements ReqContextInterface {
       this.email = user.email;
       this.userName = user.name || "";
       this.superAdmin = user.superAdmin || false;
-      this.permissions = getUserPermissions(user.id, org, teams || []);
+      this.userPermissions = getUserPermissions(user.id, org, teams || []);
     }
     // If an API key or background job is making this request
     else {
       if (!role) {
         throw new Error("Role must be provided for API key or background job");
       }
 
-      this.permissions = {
+      this.userPermissions = {
         global: {
-          permissions: roleToPermissionMap(role, org),
+          permissions: roleToPermissionMap("noaccess", org),
           limitAccessByEnvironment: false,
           environments: [],
         },
         projects: {},
       };
     }
-    this.readAccessFilter = getReadAccessFilter(this.permissions);
-  }
-
-  // Check permissions
-  public hasPermission(
-    permission: Permission,
-    project?: string | (string | undefined)[] | undefined,
-    envs?: string[] | Set<string>
-  ) {
-    return userHasPermission(
-      this.superAdmin,
-      this.permissions,
-      permission,
-      project,
-      envs ? [...envs] : undefined
-    );
-  }
+    this.readAccessFilter = getReadAccessFilter(this.userPermissions);
 
-  // Helper if you want to throw an error if the user does not have permission
-  public requirePermission(
-    permission: Permission,
-    project?: string | (string | undefined)[] | undefined,
-    envs?: string[] | Set<string>
-  ) {
-    if (!this.hasPermission(permission, project, envs)) {
-      throw new Error("You do not have permission to complete that action.");
-    }
```

Comment:
Given that hasPermission and requirePermission will be phased out, and these are not yet being consumed, I think it makes sense to remove them.

Reply:
I would add these back in for now.  The DataModel classes we're going to add soon will consume these.